thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -241,16 +235,25 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
         {
             if (builder)
             {
-                builder->finish(metadata, &crc, maxRecordSizeSeen);
-                if (!isTLK)
+                // Clear out builder before merging builder stats into inactive stats
+                // so that gatherActiveStatistics doesn't also merge builder stats.
+                Owned <IKeyBuilder> tmpBuilder;
+                {
+                    CriticalBlock b(builderCS);
+                    tmpBuilder.setown(builder.getClear());
+                }
+                if (tmpBuilder)
                 {
-                    duplicateKeyCount = builder->getDuplicateCount();
-                    numLeafNodes = builder->getNumLeafNodes();
-                    numBranchNodes = builder->getNumBranchNodes();
-                    numBlobNodes = builder->getNumBlobNodes();
-                    offsetBranches = builder->getOffsetBranches();
-                    branchMemorySize = builder->getBranchMemorySize();
-                    leafMemorySize = builder->getLeafMemorySize();
+                    tmpBuilder->finish(metadata, &crc, maxRecordSizeSeen);
+                    if (!isTLK)
+                    {
+                        duplicateKeyCount = tmpBuilder->getDuplicateCount();
+                        offsetBranches = tmpBuilder->getOffsetBranches();
+                        branchMemorySize = tmpBuilder->getBranchMemorySize();
+                        leafMemorySize = tmpBuilder->getLeafMemorySize();
+                    }
+                    mergeStats(inactiveStats, tmpBuilder, indexWriteActivityStatistics);
+                    tmpBuilder.clear();
```

Comment:
trivial: this clear() is not really needed, Owned is about to go out of scope anyway.